### PR TITLE
fix the outdated end2end training examples of moe+torchtitan

### PR DESCRIPTION
**Pull request by rakkit**
as titled.  the example [here](https://github.com/pytorch/ao/blob/main/torchao/prototype/moe_training/examples/simple_moe_layer.py)  uses some outdated `TorchTitan` APIs. This PR fixes them and aligns the example with other `torchtitan` end-to-end usages.

```
# usage for test fp8_rowwise (default use fp8_rowwise)
python ./torchao/prototype/moe_training/examples/simple_moe_layer.py 
# usage for test fp8_rowwise
python ./torchao/prototype/moe_training/examples/simple_moe_layer.py  --scaling_type fp8_rowwise

# usage for test mxfp8
python ./torchao/prototype/moe_training/examples/simple_moe_layer.py  --scaling_type mxfp8

```
results:
image

### CODE REVIEW

**Comment by rakkit** [comment]
I made a few changes to set the random seed and allow it to pass `scaling_type` from CLI.

**Comment by rakkit** [comment]
@danielvegamyhre yes, the outpout should be 
```
> python ./torchao/prototype/moe_training/examples/simple_moe_layer.py 
step 0 loss: 2656.0
step 1 loss: 2624.0
step 2 loss: 2592.0
step 3 loss: 2560.0
step 4 loss: 2528.0
step 5 loss: 2512.0
step 6 loss: 2480.0
step 7 loss: 2448.0
step 8 loss: 2432.0
step 9 loss: 2416.0
```

i have already revised the PR message and put the screenshoot there

File: torchao/prototype/moe_training/examples/simple_moe_layer.py [hunk 1]
```diff
@@ -40,14 +60,21 @@ def moe_module_filter_fn(mod: nn.Module, cur_fqn: str) -> bool:
...
+    alignment_size = 32
+
 quantize_(model, config=config, filter_fn=moe_module_filter_fn)
+set_token_group_alignment_size_m(alignment_size)
 
 # training loop
 optimizer = torch.optim.AdamW(model.parameters(), lr=1e-3)
 for step in range(10):
-    batch, seq, dim = 8, 2048, 256
+    batch, seq, dim = 8, 2048, dim
```
Comment (danielvegamyhre) [inline torchao/prototype/moe_training/examples/simple_moe_layer.py]: No reason to set `dim = dim` here, let's just do:

```python
batch, seq = 8, 2048
```

Comment (rakkit) [reply torchao/prototype/moe_training/examples/simple_moe_layer.py]: fixed, thanks.
